pyrlang/dist_proto/client.py

```diff
@@ -30,8 +31,9 @@
 class DistClientProtocol(BaseDistProtocol):
     """ Handles outgoing connections from our to other nodes. """
 
-    def __init__(self, node_name: str):
+    def __init__(self, node_name: str, dist_vsn: int):
```

Comment:
given that we know what version we want to use, could we solve it with a factory function or similar so we don't need all the if cases, or can it change mid life?

Reply:
Hi, I was on vacation...
Do you mean factory function to create specific class for a given distribution version of the protocol? With just two versions it seems reasonable but I am not sure if this strategy would have advantages if another (theoretically, as it may "never" happen) was added. Could it be considered later (next step/commit)?